src/stores/UserStoreMetamask.ts

```diff
@@ -130,6 +130,20 @@ export class UserStoreMetamask extends StoreConstructor {
     }
   }
 
+  isCorrectNetworkSelected() {
+    if (process.env.ENV === "MAINNET") {
+      if (this.chainId === '0x1' && this.network === NETWORKS.ETH) return true
+      if (this.chainId === '0x38' && this.network === NETWORKS.BSC) return true
+    }
+
+    if (process.env.ENV === "TESTNET") {
+      if ((this.chainId === '0x4' || this.chainId === '0x2a' || this.chainId === '0x3') && this.network === NETWORKS.ETH) return true
+      if (this.chainId === '0x61' && this.network === NETWORKS.BSC) return true
+    }
+
+    return false
+  }
```

Comment:
This is a bit too hardcoded. Move this to a more configurable format. Use something like `chainIdMap` to make this easily extendable using a map

Reply:
resolved on last commit